### PR TITLE
[VO-781] feat: Make helper function to copy file to either Cozy or Nextcloud

### PR DESCRIPTION
**Pull request by cballevre**

### CODE REVIEW

File: packages/cozy-stack-client/src/NextcloudFilesCollection.js [hunk 1]
```diff
@@ -226,6 +233,21 @@ class NextcloudFilesCollection extends DocumentCollection {
...
+   * Copy a file or folder to another path on the same Nextcloud server
+   *
+   * @param {object} file - The `io.cozy.remote.nextcloud.files` file to copy
+   * @param {object} to - Whether to copy the file
+   */
+  async copy(file, to) {
+    return await this.stackClient.fetchJSON(
+      'POST',
+      `/remote/nextcloud/${file.cozyMetadata.sourceAccount}/copy/${file.path}${
+        to ? `?Path=${to.path}/${file.name}` : ''
```
Comment (paultranvan) [inline packages/cozy-stack-client/src/NextcloudFilesCollection.js:247]: What happens if there is no "to" destination?

File: packages/cozy-stack-client/src/FileCollection.js [hunk 2]
```diff
@@ -440,7 +440,7 @@ class FileCollection extends DocumentCollection {
   async copy(id, name, dirId) {
     const params = {
       name: name === undefined ? undefined : sanitizeAndValidateFileName(name),
-      dirId
+      DirID: dirId
```
Comment (paultranvan) [inline packages/cozy-stack-client/src/FileCollection.js:443]: Does it mean this method never worked? Is it used somewhere?

File: packages/cozy-client/src/models/file.js [hunk 3]
```diff
@@ -654,3 +654,50 @@ export const fetchBlobFileById = async (client, fileId) => {
...
+    if (destination._type === 'io.cozy.remote.nextcloud.files') {
+      const resp = await client
+        .collection('io.cozy.remote.nextcloud.files')
+        .moveFromCozy(destination, file, { copy: true })
+      return resp
+    }
+
+    const resp = await client
+      .collection(DOCTYPE_FILES)
+      .copy(file._id, undefined, destination._id)
```
Comment (paultranvan) [inline packages/cozy-client/src/models/file.js:698]: We don't want to pass `file.name`?

**Review by paultranvan — APPROVED** [review]

File: packages/cozy-stack-client/src/NextcloudFilesCollection.js (hunk 1 above)
Comment (cballevre) [reply packages/cozy-stack-client/src/NextcloudFilesCollection.js:247]: The file will be copied into the same folder as the file being copied

File: packages/cozy-stack-client/src/FileCollection.js (hunk 2 above)
Comment (cballevre) [reply packages/cozy-stack-client/src/FileCollection.js:443]: This was not used by Drive where there was duplicate function. I follow the [documentation](https://docs.cozy.io/en/cozy-stack/files/#post-filesfile-idcopy) after realising on Nextcloud that the stack was sensitive to the case in these arguments.

File: packages/cozy-client/src/models/file.js (hunk 3 above)
Comment (cballevre) [reply packages/cozy-client/src/models/file.js:698]: Because I want the stack to manage the renaming to avoid conflicts

**Pull request merged**